src/Bicep.Core.UnitTests/Diagnostics/LinterRuleTests/SecureParamsInNestedDeploymentsTests.cs

```diff
@@ -0,0 +1,749 @@
+                              type: 'Microsoft.Storage/storageAccounts'
+                              apiVersion: '2021-04-01'
+                              #disable-next-line no-loc-expr-outside-params
+                              location: format('{0}', listKeys('someResourceId', '2020-01-01'))
+                              kind: 'StorageV2'
+                              sku: {
+                                name: 'Premium_LRS'
+                                tier: 'Premium'
+                              }
+                            }
+                          ]
+                        }
+                      }
+                    }",
+                new string[]
+                {
+                });
+        }
+
+        [TestMethod]
+        public void ExplicitlyOuterScope_ButNoSecureParams_OrListKeys_Pass()
+        {
+            // https://docs.microsoft.com/en-us/azure/azure-resource-manager/templates/template-test-cases#use-inner-scope-for-nested-deployment-secure-parameters
+            CompileAndTest(@"
+                    param stgAccountName string
+
+                    resource nested 'Microsoft.Resources/deployments@2021-04-01' = {
+                      name: 'nested'
+                      properties: {
+                        mode: 'Incremental'
+                        expressionEvaluationOptions: {
+                          scope: 'outer'
+                        }
+                        template: {
+                          '$schema': 'https://schema.management.azure.com/schemas/2019-04-01/deploymentTemplate.json#'
+                          contentVersion: '1.0.0.0'
+                          resources: [
+                            {
+                              name: stgAccountName
+                              type: 'Microsoft.Storage/storageAccounts'
+                              apiVersion: '2021-04-01'
+                              #disable-next-line no-loc-expr-outside-params
+                              location: resourceGroup().location
+                              kind: 'StorageV2'
+                              sku: {
+                                name: 'Premium_LRS'
+                                tier: 'Premium'
+                              }
+                            }
+                          ]
+                        }
+                      }
+                    }",
+                new string[]
+                {
+                });
+        }
+
+        [TestMethod]
+        public void ImplicitlyOuterScope_ButNoSecureParams_OrListKeys_Pass()
+        {
+            // https://docs.microsoft.com/en-us/azure/azure-resource-manager/templates/template-test-cases#use-inner-scope-for-nested-deployment-secure-parameters
+            CompileAndTest(@"
+                    param stgAccountName string
+
+                    resource nested 'Microsoft.Resources/deployments@2021-04-01' = {
+                      name: 'nested'
+                      properties: {
```

Comment:
Scope is not mentioned here. Is it the default?

Reply:
Yes.  That's why the test is called "ImplicitlyOuterScope". Adding comment.